spec/helpers/storage_helper/textual_summary_spec.rb

```diff
@@ -18,4 +18,61 @@
       expect(textual_format_used_space(3331, 10, 1)).to eq('3.25 KB (10% of Used Space, 1 file)')
     end
   end
+
+  describe '#textual methods' do
+    context 'for zero VMs' do
+      before do
+        @record = FactoryGirl.create(:storage)
+      end
+
+      it 'returns correct Hash' do
+        expect(textual_registered_vms).to eq(:label => "Managed/Registered VMs",
+                                             :icon  => "pficon pficon-virtual-machine",
+                                             :value => 0)
+      end
+      it 'returns correct Hash' do
+        expect(textual_unregistered_vms).to eq(:label => "Managed/Unregistered VMs",
+                                               :icon  => "pficon pficon-virtual-machine",
+                                               :value => 0)
+      end
+      it 'returns correct Hash' do
+        expect(textual_unmanaged_vms).to eq(:label => "Unmanaged VMs",
+                                            :title => "Unmanaged VMs are no longer available",
+                                            :icon  => "pficon pficon-virtual-machine",
+                                            :value => 0)
+      end
+    end
+
+    context 'for any number of VMs' do
+      before do
+        @record = FactoryGirl.create(:storage)
+        allow_any_instance_of(described_class).to receive(:role_allows?).and_return(true)
```

Comment:
@skateman Any ideas how to make it better? I need to mock those methods but in nicer way.

Reply:
What about `subject`?

Reply:
or `self`

Reply:
`self` works. Thanks 👍